From e3c8de6752fa9c53d32521553f47f49f79f772b8 Mon Sep 17 00:00:00 2001
From: jzhuan5 <jin.can.zhuang@intel.com>
Date: Fri, 17 Aug 2012 14:00:18 -0400
Subject: [PORT FROM MAIN] usb:adbd:ignore usb service if already in usb mode

BZ: 51645

restarting adbd if usb is already in usb mode causes adbd in
offline state, and thus should be avoid.
Signed-off-by: jzhuan5 <jin.can.zhuang@intel.com>

Change-Id: I345544962c16a3d8ab52da65c9fa3eb437c1d92b
Orig-Change-Id: Ied468f5f64c83dc3bd668c7b711d9ca4dd0b78b7
Reviewed-on: http://android.intel.com:8080/77187
Reviewed-by: Zhuang, Jin Can <jin.can.zhuang@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Meng, Zhe <zhe.meng@intel.com>
Tested-by: Meng, Zhe <zhe.meng@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 adb/services.c |    9 +++++++++
 adb/sockets.c  |    6 +++++-
 2 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/adb/services.c b/adb/services.c
index 54d21a8..f0318c6 100644
--- a/adb/services.c
+++ b/adb/services.c
@@ -155,6 +155,15 @@ void restart_tcp_service(int fd, void *cookie)
 void restart_usb_service(int fd, void *cookie)
 {
     char buf[100];
+    char value[PROPERTY_VALUE_MAX];
+
+    property_get("service.adb.tcp.port", value, "0");
+    if (!strcmp(value, "0")) {
+        snprintf(buf, sizeof(buf), "already in USB mode\n");
+        writex(fd, buf, strlen(buf));
+        adb_close(fd);
+        return;
+    }
 
     property_set("service.adb.tcp.port", "0");
     snprintf(buf, sizeof(buf), "restarting in USB mode\n");
diff --git a/adb/sockets.c b/adb/sockets.c
index 305cb44..5b56ed3 100644
--- a/adb/sockets.c
+++ b/adb/sockets.c
@@ -411,6 +411,9 @@ asocket *create_local_service_socket(const char *name)
 {
     asocket *s;
     int fd;
+#if !ADB_HOST
+    char value[PROPERTY_VALUE_MAX];
+#endif
 
 #if !ADB_HOST
     if (!strcmp(name,"jdwp")) {
@@ -427,8 +430,9 @@ asocket *create_local_service_socket(const char *name)
     D("LS(%d): bound to '%s' via %d\n", s->id, name, fd);
 
 #if !ADB_HOST
+    property_get("service.adb.tcp.port", value, "0");
     if ((!strncmp(name, "root:", 5) && getuid() != 0)
-        || !strncmp(name, "usb:", 4)
+        || (!strncmp(name, "usb:", 4) && strcmp(value, "0"))
         || !strncmp(name, "tcpip:", 6)) {
         D("LS(%d): enabling exit_on_close\n", s->id);
         s->exit_on_close = 1;
-- 
1.7.4.1

